### PR TITLE
105707 Convert tie xml to TiObject

### PR DESCRIPTION
**Pull request by jsoi-eq**
[AB#105707](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/105707)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            $"Error when committing message. Exception: {e.Message} Stacktrace: {e.StackTrace} TIEMessage: {message}",
+            e.Message, e.StackTrace, message);
+
+        return tiMessageResult;
+    }
+
+    private static void AddResultOfImportOperationToResponseObject(TIInterfaceMessage message, TIMessageResult? tiMessageResult,
+        TIResponseFrame response)
+    {
+        if (tiMessageResult != null)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs:128]: is not

File: src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        _importSchemaMapperMock = Substitute.For<IImportSchemaMapper>();
+        _loggerMock = Substitute.For<ILogger<ImportHandler>>();
+        _dut = new ImportHandler(_importSchemaMapperMock, _loggerMock);
+    }
+
+    [TestMethod]
+    public void Handle_ShouldReturnErrorResultIfMapperFails()
+    {
+        // Arrange
+        var mappingResultNotSuccess = new MappingResult(new TIMessageResult{ErrorMessage = "Mapping failed", Result = MessageResults.Failed});
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs:27]: long line

File: src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+using NSubstitute;
+using Statoil.TI.InterfaceServices.Message;
+
+namespace Equinor.ProCoSys.Completion.TieImport.Tests;
+
+[TestClass]
+public class ImportHandlerTests
+{
+    IImportSchemaMapper _importSchemaMapperMock;
+    ILogger<ImportHandler> _loggerMock;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs:12]: I guess your not planning to assert any stuff on logging, so you don't need _loggerMock as global.
Throw it and do _dut = new ImportHandler(_importSchemaMapperMock, Substitute.For<ILogger<ImportHandler>>());

File: src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        _importSchemaMapperMock.Map(Arg.Any<TIInterfaceMessage>()).Returns(mappingResultNotSuccess);
+
+        //Act
+        var result = _dut.Handle(new TIInterfaceMessage());
+
+        //Assert
+        Assert.AreEqual(MessageResults.Failed, result.Results[0].Result);
+        Assert.AreEqual("Mapping failed", result.Results[0].ErrorMessage);
+    }
+}
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport.Tests/ImportHandlerTests.cs:38]: Also add a minor positve test?

File: src/Equinor.ProCoSys.Completion.TieImport/Adapter/Tie1MessageHandler.cs [hunk 5]
```diff
@@ -10,11 +10,13 @@ public class Tie1MessageHandler : IMessageHandler<TieAdapterConfig, TieAdapterPa
 {
     private readonly ILogger<Tie1MessageHandler> _logger;
     private readonly IImportSchemaMapper _commonLibMapper;
+    private readonly IImportHandler _importHandler;
 
-    public Tie1MessageHandler(ILogger<Tie1MessageHandler> logger, IImportSchemaMapper commonLibMapper)
+    public Tie1MessageHandler(ILogger<Tie1MessageHandler> logger, IImportSchemaMapper commonLibMapper, IImportHandler importHandler)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport/Adapter/Tie1MessageHandler.cs:15]: Add some minimum positive tests

File: src/Equinor.ProCoSys.Completion.TieImport/Equinor.ProCoSys.Completion.TieImport.csproj [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
   </PropertyGroup>
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport/Equinor.ProCoSys.Completion.TieImport.csproj:7]: Add `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`

File: src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+        //TODO: 107052 Create command and send to command handler
+
+        //TODO: 106687 CommandFailureHandler;
+
+        //TODO: 109642 return ImportResult.Ok();
+    }
+
+
+    private TIMessageResult? HandleExceptionFromImportOperation(TIInterfaceMessage message, Exception e)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs:115]: remove nullable on return type

File: src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        var tiMessageResult = e.ToMessageResult();
+        _logger.LogError(
+            $"Error when committing message. Exception: {e.Message} Stacktrace: {e.StackTrace} TIEMessage: {message}",
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs:119]: not string interpolation, but
image
Search through other logging statements and do the same

File: src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs:128]: Consider to make tiMessageResult not nullable and throw the if

File: src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs [hunk 9]
```diff
@@ -21,19 +22,19 @@ public static class TieImportModule
...
+            .ValidateDataAnnotations();
         configuration.Bind("TieImport", configOptions);
 
         services.AddOptions<CommonLibOptions>()
             .BindConfiguration("CommonLib")
-            .ValidateDataAnnotations()
-            .ValidateOnStart();
+            .ValidateDataAnnotations();
 
+        //TODO: JSOI Scoped or Singleton or Transient?
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs:35]: we can discuss this in team .. I've never been sure on this myself ..

File: src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs:1]: on line 68, discard unused params c+ ex
image

Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs:1]: On line 118: //TODO: JSOI - Figure out how to get logger object
-> Add a number to a work item

File: src/Equinor.ProCoSys.Completion.WebApi/Equinor.ProCoSys.Completion.WebApi.csproj [hunk 10]
```diff
@@ -61,4 +64,9 @@
       <Link>.dockerignore</Link>
     </Content>
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Equinor.ProCoSys.Completion.WebApi.csproj:68]: remove (empty folder)

File: src/Equinor.ProCoSys.Completion.WebApi/appsettings.json
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/appsettings.json:1]: Undo changes in this file .. has been changed to other setting lately

File: src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs (hunk 1 above)
Comment (jsoi-eq) [reply src/Equinor.ProCoSys.Completion.TieImport/ImportHandler.cs:128]: I think if the code crashes on "var mapped = _importSchemaMapper.Map(message);" statement, code will jump to catch and then finally clause.

Finally clause calls AddResultOfImportOperationToResponseObject, and if tiMessageResult is uninitialized we cannot set the Guid and ExternalReference properties.

So I don't think I can remove the nullable.

File: src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs (hunk 9 above)
Comment (jsoi-eq) [reply src/Equinor.ProCoSys.Completion.WebApi/DiModules/TieImportModule.cs:35]: Yes please...

**Review by janid1967 — APPROVED** [review]

**Pull request merged**